src/java_bytecode/java_types.h

```diff
@@ -313,6 +319,66 @@ inline const typet &java_generic_class_type_bound(size_t index, const typet &t)
   return gen_type.subtype();
 }
 
+/// Type to hold a Java class that is implicitly generic, e.g., an inner
+/// class of a generic outer class or a derived class of a generic base
+/// class. Extends the java class type.
+class java_implicitly_generic_class_typet:public java_class_typet
+{
+public:
+  typedef std::vector<java_generic_parametert> implicit_generic_typest;
+
+  static const java_implicitly_generic_class_typet &from_java_class_type(
+    java_class_typet &class_type,
+    implicit_generic_typest implicit_generic_types)
+  {
+    class_type.set(ID_C_java_implicitly_generic_class_type, true);
```

Comment:
Return something derived rather than altering in place

Reply:
This function has been now removed, and I use a constructor to do the job instead.